### ISSUE TITLE
/capabilities hosts do not honour port number

### ISSUE THREAD

**Issue by at88mph** [opencadc/reg#12]
For those users that run the service directly on port 8080, for example, the /capabilities will return the hostname without the port number.

**Comment by brianmajor** [comment]
Dustin,
INAF encountered this problem too and may have fixed.  I'll ask.
Brian

**Comment by pdowler** [comment]
The current code processes a template and replaces the server name with the one in the request. Are we expecting that it should replace the server name in the template with server:port? Or should the service just put the port number in the document if that is how they deploy stuff? I realise the latter means the app has to know the deployment environment (which port the front end http server is listening on) but it already does make assumptions that the apache/tomcat config can effect....

on replacing server namer with server:port -- that would not really work easily because http and https are on different ports and both kinds of urls are reported in the output doc.

**Comment by at88mph** [comment]
Do service implementors need to edit the `capabilities.xml` file to reflect port changes, then?  

The issue is that there is an assumption that Apache is fronting Tomcat, which isn't always the case.  In the case where our services are containerized, for example, the ports will probably just be 8080 when they interact with each other.

**Comment by pdowler** [comment]
The output of /capabilities has to provide URLs that will work for the caller so one has to know how the service is deployed to provide that document. The server-name rewriting now is just to support development and rc testing. We are not assuming apache is in front of tomcat: I have stand-alone tomcat on port 80/443 using jvc on my desktop and this also works here. It is using knowledge of the ports because it has to know that to provide values for the accessURL elements (in our case, we know that default ports for those protocols are used). So yeah, this particular feature is hard to implement in a deloyment-agnostic fashion.

**Comment by at88mph** [comment]
OK, but for an immediate deployment, is one expected to modify the `capabilities.xml` for a particular service, given that knowledge of the port that it will deployed on is a requirement?

**Comment by pdowler** [comment]
Yes, that would be the only way to make the app work on another deployment.
